library/Fum2GitHub/GitHub.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Fum2GitHub.GitHub (
+    getNextUrl,
+    getOrgMembers,
+    OAuthToken(OAuthToken),
+    OrgMember(getOrgMember),
+    URL(URL),
+) where
+
+import           Control.Applicative ((<$>))
+import           Data.Aeson ((.:))
+import qualified Data.Aeson as Aeson
+import           Data.Aeson.Types (parseEither)
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as E
+import           Data.Traversable (traverse)
+import qualified Data.Vector as Vector
+import           Network.HTTP.Conduit (
+    applyBasicAuth,
+    httpLbs,
+    parseUrl,
+    requestHeaders,
+    responseBody,
+    responseHeaders,
+    withManager)
+import           Network.HTTP.Types.Header (ResponseHeaders)
+import           Text.Regex (mkRegex, matchRegex)
+
+
+newtype URL = URL { getURL :: String } deriving (Eq, Show)
+
+newtype OAuthToken = OAuthToken { getOAuthToken :: String }
+
+
+-- Get the response body and headers from url using oAuthToken.
+getHttp :: URL -> OAuthToken -> IO (LBS.ByteString, ResponseHeaders)
```

Comment:
The `getHttp`s in Fum.hs and GitHub.hs are a bit different now (they add different headers and one also returns the ResponseHeaders).

But this suggested a bigger refactor, in a separate PR, which puts `getHttp` and traversing all API pages in a common module:
```hs
-- Util.hs
getHttp :: Request -> Response

getAllJSON :: Request -> (Response -> Maybe Request) -> (Response -> Either String [a]) -> IO Either String [a]
getAllJSON initialReq getNextReq getResults =
  concatenate the results from all API pages
```
Fum/GitHub have different code for getting the next URL (http headers vs json body) and JSON results (different paths in the response body) but duplicate the logic to get the next page of results and concatenate them.

Unless such a refactor would over-complicate things?